### ISSUE TITLE
Gamepad buttons don't work

### ISSUE THREAD

**Issue by memegod4252** [hrydgard/ppsspp#18574]
### Game or games this happens in

God of War ghost of sparta

### What area of the game / PPSSPP

The circle and square button  and the analog up and down don't work on the psp emulator i even tried it on other games like shin bou do kai but the problem isnt from the controller it works on other games outside of the psp emualator and i even tested it online and everything's fine i even tested it on my brothers psp and it worked.Also the keyboard controls work. Idk i think the issue is because i use mac and not windows if anyone knowns pls tell me

### What should happen

Every button should work

### Logs

![Screen Shot 2023-12-18 at 4 07 32 PM](https://github.com/hrydgard/ppsspp/assets/154247525/698b3b4d-75ea-4edd-9afa-e6530ca2cc48)

### Platform

macOS

### Mobile phone model or graphics card

Intel HD Graphics 6000 1536 MB

### PPSSPP version affected

v1.16.5

### Last working version

v1.16.5

### Graphics backend (3D API)

Other

### Checklist

- [X] Test in the [latest git build](https://buildbot.orphis.net/ppsspp/) in case it's already fixed.
- [X] [Search for other reports](https://github.com/hrydgard/ppsspp/search?q=my+issue&type=issues) of the same issue.
- [X] Try resetting settings or older versions and include if the issue is related.
- [X] Try without any cheats and without loading any save states.
- [X] Include logs or screenshots of issue.

**Comment by hrydgard** [comment]
It looks to me like you simply haven't bound square and circle? are you unable to bind them?

**Comment by memegod4252** [comment]
sorry i don't relayy know what binding is can you please explain and also how do i bind them

**Comment by hrydgard** [comment]
Binding is just setting up which of your buttons will affect which button on the PSP. Click on the + next to circle, then press the button you want to be circle. Use the little garbage cans to delete bindings you don't need.

**Comment by memegod4252** [comment]
yes i have tried binding but when i click circle it doesnt registervit's like the button is not there it doesn't read it same as analog up and down

**Comment by memegod4252** [comment]
when i try to bind those buttons nothing happens it just keeps saying map a new key for circle

**Comment by memegod4252** [comment]
Please if u have a solution to fix this i would really appreciate it Thx.

**Comment by memegod4252** [comment]
I'm unable to bind the buttons, do you have a solution?

**Comment by hrydgard** [comment]
I don't know, this sounds very weird.

Can you try this? In PPSSPP, go to Settings / Tools / Developer Tools / Touchscreen Test (it tests more than touchscreens). Then press the buttons that don't seem to be able to map, and if something shows on the screen when you do, take a screenshot and post here.

**Comment by memegod4252** [comment]
Unfortunatly nothing shows up it's like ppsspp doesn't even recognise the buttons.

**Comment by hrydgard** [comment]
What controller are you using? Maybe we got a bad entry for it in assets/gamecontrollerdb.txt , which is used on Mac and Linux to map the real buttons to the usual xbox-like virtual buttons

**Comment by memegod4252** [comment]
i think that may be the problem because my controller doesn't hve X or circle it has 2 and 3.

Here's a picture.

![Uploading Screen Shot 2023-12-19 at 3.20.58 PM.png…]()

**Comment by hrydgard** [comment]
I'm not saying you did, all I'm saying is that some controllers work on Mac.

I just tried connecting a PS4 controller via USB to my Mac, and all the buttons were automatically correctly mapped except L and R, which were easily manually fixed. 

If you press "Auto configure" on the control mappings screen, what's the name and number of the controller it shows there? Can you take a screenshot of that?

**Comment by memegod4252** [comment]
Sure.

![Uploading Screen Shot 2023-12-19 at 3.23.05 PM.png…]()

**Comment by memegod4252** [comment]
When i clicked auto configure it selects the buttons, but when you go in game still nothing happens when i click circle.

**Comment by hrydgard** [comment]
I don't know what happened but your screenshots cannot be viewed. Just drag the image files into your post and wait until it's uploaded before you post.

**Comment by memegod4252** [comment]
![Screen Shot 2023-12-19 at 3 26 31 PM](https://github.com/hrydgard/ppsspp/assets/154247525/a67a5a23-db2a-4927-826e-b6188be12139)

**Comment by memegod4252** [comment]
Also i would like to add something on Mac, the select button says pad1.Touch Major+, but on the windows it says pad1. b9

**Comment by hrydgard** [comment]
What the buttons correspond to are controlled by the file `assets/gamecontrollerdb.txt`. We don't seem to have an entry for your controller there.

So the fix for all this is likely to find the correct configuration line and put it there.

In https://github.com/hrydgard/ppsspp/issues/18091 we link to a regularly updated one, but it doesn't contain a line for your controller, for whatever reason.

Also, googling for "03006ce8100800000100000006010000" didn't help, though I might have made a mistake there.

I'm not really sure why SDL needs these configurations to read controllers correctly, but it does.

**Comment by memegod4252** [comment]
Do you know why it work on windows but not mac?

**Comment by hrydgard** [comment]
On Windows we use DirectInput and XInput, not SDL, for joystick input. They work differently..

**Comment by memegod4252** [comment]
I have actually found a solution to this problem, i downloaded a software wich links my controller keys to my keynoard keys then it registers them in the game beacuse my controller keys don't work and the keyboard keys do.

**Comment by hrydgard** [comment]
OK, good that you find a workaround. Still, this is quite weird.
